test/testcommon/src/main/java/io/pravega/test/common/TestUtils.java

```diff
@@ -41,7 +41,7 @@
      *
      * @return free port.
      */
-    public static int getAvailableListenPort() {
+    public synchronized static int getAvailableListenPort() {
```

Comment:
This change seems to be gratuitous.

Reply:
This method is to get a free port but without synchronization two threads may call at same time and get same port which eventually cause port collision. I experienced this problem once.

Reply:
And removed this synchronization to a separate issue/PR, per the conversation with Flavio.